Polyhedron/demo/Polyhedron/resources/shader_flat.f

```diff
@@ -13,11 +14,20 @@
 uniform vec4 light_spec;
 uniform vec4 light_amb;
 uniform float spec_power ;
+uniform bool is_clipbox_on;
 
 out vec4 out_color;
 
 void main(void)
 {
+  if(is_clipbox_on)
+    if(fs_in.dist[0]>0 ||
+       fs_in.dist[1]>0 ||
+       fs_in.dist[2]>0 ||
+       fs_in.dist[3]>0 ||
+       fs_in.dist[4]>0 ||
+       fs_in.dist[5]>0)
+      discard;
```

Comment:
You should bench that. Adding a `if` in a shader might penalize the performance of the shader, even for items when the `is_clipbox_on` is always false. If the performance is impacted, you should use two different shaders.

Reply:
That is easier said than done. That would mean that all impacted items should test for the presence of an active clipping box  in the scene, qhich they cannot do because they do not have a reference to the scene, and choose their program accordingly. So just for that little plugin we need to add a boolean to the Scene_facegraph_items and 2 programs to the list in the viewer. 
I agree that it is probably impacting the rendering, but the alternative does not satisfy me either.

Reply:
I benched it, it doesn't change a thing.